### PR TITLE
Add About command for help menu

### PR DESCRIPTION
**Pull request by bigfluffycookie**
https://github.com/SonarSource/sonarlint-visualstudio/issues/3371

aboutCommand

### CODE REVIEW

File: src/Integration.Vsix/Commands/PackageCommandId.cs [hunk 1]
```diff
@@ -31,6 +31,7 @@ internal enum PackageCommandId
         // Menus
         ProjectSonarLintMenu = 0x850,
         // Help menu buttons
-        SonarLintHelpShowLogs = 0x102
+        SonarLintHelpShowLogs = 0x102,
+        AboutCommand = 0x104
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandId.cs:35]: Same comment regarding the ID

File: src/Integration.Vsix/Commands/PackageCommandManager.cs [hunk 2]
```diff
@@ -44,7 +46,8 @@ public PackageCommandManager(IMenuCommandService menuService)
         public void Initialize(ITeamExplorerController teamExplorerController,
             IProjectPropertyManager projectPropertyManager,
             IProjectToLanguageMapper projectToLanguageMapper,
-            IOutputWindowService outputWindowService)
+            IOutputWindowService outputWindowService,
+            IBrowserService browserService)
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandManager.cs:50]: I don't like that this class gets the dependencies of all the commands it initializes.
We might need to revisit it later, once we have more use cases.

File: src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Commands
+{
+    internal class AboutCommand : VsCommandBase
+    {
+        private readonly string aboutUrl = "https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode";
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs:12]: LEYLA!!!
For shame.

image

File: src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs:1]: copyright headers

File: src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        [TestMethod]
+        public void Invoke_VerifyCallsBrowserServiceNavigate()
+        {
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+
+            var browserService = new Mock<IBrowserService>();
+
+            var testSubject = new AboutCommand(browserService.Object);
+
+            var aboutUrl = "https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode";
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs:41]: For shame.

**Review by rita-sonarsource — CHANGES_REQUESTED** [review]
See my comments. Tsk tsk tsk.

File: src/Integration.Vsix/Commands/PackageCommandManager.cs (hunk 2 above)
Comment (bigfluffycookie) [reply src/Integration.Vsix/Commands/PackageCommandManager.cs:50]: Yea I agree, its starting to get ugly

File: src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs [hunk 6]
```diff
@@ -86,10 +86,11 @@ public void PackageCommandManager_Initialize()
             testSubject.Initialize(serviceProvider.GetMefService<ITeamExplorerController>(),
                 serviceProvider.GetMefService<IProjectPropertyManager>(),
                 Mock.Of<IProjectToLanguageMapper>(),
-                Mock.Of<IOutputWindowService>());
+                Mock.Of<IOutputWindowService>(),
+                Mock.Of<IBrowserService>());
 
             // Assert
-            menuService.Commands.Should().HaveCount(allCommands.Count, "Unexpected number of commands");
+            menuService.Commands.Should().HaveCountGreaterOrEqualTo(allCommands.Count, "Unexpected number of commands");
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs:93]: Could you explain why this change was needed?

File: src/Integration.Vsix/Commands/PackageCommandManager.cs [hunk 7]
```diff
@@ -58,6 +61,7 @@ public PackageCommandManager(IMenuCommandService menuService)
 
             // Help menu buttons
             this.RegisterCommand(CommonGuids.HelpMenuCommandSet, (int)PackageCommandId.SonarLintHelpShowLogs, new ShowLogsCommand(outputWindowService));
+            this.RegisterCommand(CommonGuids.HelpMenuCommandSet, AboutCommand.AboutCommandId, new AboutCommand(browserService));
```
Comment (georgii-borovinskikh-sonarsource) [inline src/Integration.Vsix/Commands/PackageCommandManager.cs:64]: It's different from the other commands, but I like it this way more. Should we discuss making that to be the way of defining the ids for future commands?

File: src/Integration.Vsix/Commands/HelpMenu/HelpMenu.vsct [hunk 8]
```diff
@@ -15,16 +15,31 @@
...
+          <Parent guid="HelpMenuCmdSet" id="HelpGroup" />
+          <Icon guid="guidImages" id="sonarlint" />
+          <Strings>
+            <ButtonText>About SonarLint</ButtonText>
+          </Strings>
+        </Button>
     </Buttons>
+
+    <Bitmaps>
+      <Bitmap guid="guidImages" href="..\..\Resources\sonarlint.png"/>
```
[the remark this answers is not in the document]
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Commands/HelpMenu/HelpMenu.vsct:28]: +1

**Review by georgii-borovinskikh-sonarsource — CHANGES_REQUESTED** [review]

File: src/Integration.Vsix/Commands/PackageCommandManager.cs (hunk 7 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Commands/PackageCommandManager.cs:64]: Found a possible issue with this approach in the other PR

File: src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs (hunk 6 above)
Comment (bigfluffycookie) [reply src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs:93]: As I moved the commandId away from PackageCommandId.cs this test was failing as it was comparing the amount of commands there with the commands registered.

File: src/Integration.Vsix/Commands/PackageCommandManager.cs (hunk 7 above)
Comment (bigfluffycookie) [reply src/Integration.Vsix/Commands/PackageCommandManager.cs:64]: Yes, we will do it this way now I changed it in my prs but I didnt change it in the already merged one as that should be a seperate pr :)

File: src/Integration.Vsix/Commands/HelpMenu/HelpMenu.vsct (hunk 8 above)
[the remark this answers is not in the document]
Comment (bigfluffycookie) [reply src/Integration.Vsix/Commands/HelpMenu/HelpMenu.vsct:28]: The standard for visual studio icons is 16x16 which is why i am using this one and not the 32x32

File: src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs (hunk 6 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs:93]: I think we need to find a way to keep this test precise, avoid other issues of defining the Id as a const field of a class, but also move away from using enums.

File: src/Integration.Vsix/Commands/PackageCommandManager.cs (hunk 7 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/Commands/PackageCommandManager.cs:64]: Could you look at my comments on the other PR, I found an issue with this approach

File: src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs (hunk 6 above)
Comment (bigfluffycookie) [reply src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs:93]: @rita-sonarsource thoughts?

File: src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        [TestMethod]
+        public void Invoke_VerifyCallsBrowserServiceNavigate()
+        {
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
+
+            var browserService = new Mock<IBrowserService>();
+
+            var testSubject = new AboutCommand(browserService.Object);
+
+            var aboutUrl = "https://marketplace.visualstudio.com/items?itemName=SonarSource.SonarLintforVisualStudio2022";
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs:41]: const string

File: src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Commands
+{
+    [TestClass]
+    public class AboutCommandTests
+    {
+        [TestMethod]
+        public void Invoke_VerifyCallsBrowserServiceNavigate()
+        {
+            OleMenuCommand command = CommandHelper.CreateRandomOleMenuCommand();
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/Commands/HelpMenu/AboutCommandTests.cs:35]: var ?

File: src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs (hunk 6 above)
Comment (rita-sonarsource) [reply src/Integration.Vsix.UnitTests/Commands/PackageCommandManagerTests.cs:93]: I think we should stop using PackageCommandManager as a service that knows of everything, and that will solve the problem. It should have a generic "Register command" service that the other commands can call, not the other way around. I wouldn't change it now though.

File: src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Commands
+{
+    internal class AboutCommand : VsCommandBase
+    {
+        internal const int AboutCommandId = 0x104;
+
+        private readonly string aboutUrl = "https://marketplace.visualstudio.com/items?itemName=SonarSource.SonarLintforVisualStudio2022";
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/Commands/HelpMenu/AboutCommand.cs:29]: Ah, small problem here, if this is running in VS 2019 then technically the market page is incorrect. I don't have a solution for this, a better way would be to show an About screen and we already have a ticket for this. Let's leave it for now.

File: src/Integration.Vsix/Commands/PackageCommandManager.cs (hunk 7 above)
Comment (rita-sonarsource) [reply src/Integration.Vsix/Commands/PackageCommandManager.cs:64]: @georgii-borovinskikh-sonarsource can you link to it?

**Review by rita-sonarsource — APPROVED** [review]
